src/_stories/IdleTimeout.stories.js

```diff
@@ -0,0 +1,98 @@
+import { number, text, withKnobs } from '@storybook/addon-knobs'
+import { storiesOf } from '@storybook/react'
+import { Fragment, useEffect, useRef } from 'react'
+import { div, h, span } from 'react-hyperscript-helpers'
+import { ButtonPrimary } from 'src/components/common'
+import IdleTimeoutModal from 'src/components/IdleTimeoutModal'
+import Modal from 'src/components/Modal'
+import { getDynamic, listenDynamic, setDynamic } from 'src/libs/browser-storage'
+import { authStore } from 'src/libs/state'
+import * as Utils from 'src/libs/utils'
+
+
+const dynamicStorageSlot = (storage, key) => {
+  const { subscribe, next } = Utils.subscribable()
+  const get = () => getDynamic(storage, key)
+  const set = newValue => {
+    setDynamic(storage, key, newValue)
+    next(newValue)
+  }
+  listenDynamic(storage, key, next)
+  return { subscribe, get, set, update: fn => set(fn(get())) }
+}
+
+const authTest = dynamicStorageSlot(localStorage, 'auth-story')
+authTest.update(v => v || {})
+const agreeTest = dynamicStorageSlot(localStorage, 'terra-timeout-agree')
+
+const Container = ({ modal }) => {
+  const userEmail = text('User email domain', 'foo.com')
+  const agree = Utils.useStore(agreeTest)
+  const auth = Utils.useStore(authTest)
+  const { isSignedIn } = auth
+  const modalRef = useRef()
+
+  agree && authStore.set(auth)
+
+  useEffect(() => {
+    authTest.set({ user: { id: `${userEmail}-123` }, isSignedIn: false, profile: { email: `user@${userEmail}` } })
+  }, [userEmail])
+
+  Utils.useOnMount(() => {
+    agreeTest.set(false)
+
+    const observer = new MutationObserver(() => {
+      const logout = document.evaluate('//iframe[@src="https://www.google.com/accounts/Logout"]',
+        document.body, null, XPathResult.FIRST_ORDERED_NODE_TYPE, null).singleNodeValue
+      logout && authTest.set({ user: { id: `${userEmail}-123` }, profile: {} })
+    })
+
+    observer.observe(document.body, { attributes: false, childList: true, subtree: true })
+    return () => { observer.disconnect() }
+  })
+
+  const agreement = h(Fragment, [
+    span({ style: { width: 300 } }, [
+      'Running this story will log you out of all of your google accounts in this browser.',
```

Comment:
I was on the fence about this story. 
You _may_ get logged out. This story checks for the existence of the logout iframe and mocks the authstore to log the user out which will immediately unmount the iframe. Depending on how long this takes this may log you out.

In the real world case the iframe will not unmount until the user has actually been signed out as it uses the authstore state to determine if the user has signed out.